src/store.rs

```diff
@@ -39,21 +40,23 @@ impl InitialStorage for InitialStorageMemory {
 }
 
 pub trait ContractStorage: Debug {
-    fn decommit(&self, hash: U256) -> Result<Option<Vec<U256>>, StorageError>;
-    fn hash_map(&self) -> Result<HashMap<U256, Vec<U256>>, StorageError>;
+    fn decommit(&mut self, hash: U256) -> Result<Option<Vec<U256>>, StorageError>;
+    fn decommited_hashes(&self) -> Result<HashSet<U256>, StorageError>;
```

Comment:
I'd suggest the following methods instead:
```suggestion
    fn decommit(&'a mut self, hash: U256) -> Result<Option<&'a [U256]>, StorageError>;
    fn contains_decommited_hash(&self, hash: U256) -> Result<bool, StorageError>;
```